File: be2-scala/src/main/scala/ch/epfl/pop/model/network/method/Broadcast.scala
package ch.epfl.pop.model.network.method

import ch.epfl.pop.json.HighLevelProtocol._
import ch.epfl.pop.model.network.Parsable
import ch.epfl.pop.model.network.method.message.Message
import ch.epfl.pop.model.objects.Channel
import spray.json._

final case class Broadcast(override val channel: Channel, override val message: Message) extends ParamsWithMessage(channel, message)

object Broadcast extends Parsable {
  def apply(channel: Channel, message: Message): Broadcast = {
    new Broadcast(channel, message)
  }

  override def buildFromJson(payload: String): Broadcast = payload.parseJson.asJsObject.convertTo[Broadcast]
}

File: be1-go/message/query/method/broadcast.go
package method

import (
	"popstellar/message/query"
	"popstellar/message/query/method/message"
)

// Broadcast ....
type Broadcast struct {
	query.Base

	Params struct {
		Channel string          `json:"channel"`
		Message message.Message `json:"message"`
	} `json:"params"`
}
